core/envelope-sim/src/main/java/fr/sncf/osrd/envelope_sim/EnvelopeSimPath.java

```diff
@@ -183,6 +217,12 @@ static <T> RangeMap<Double, T> mergeRanges(RangeMap<Double, T> map) {
         return result;
     }
 
-    /** The catenary electrification mode and the electrical profile level (if any) present at some position */
-    public static final record ModeAndProfile(String mode, String profile) {}
+    /**
+     * Electrification conditions at a point in the path:
+     * - catenary mode
+     * - electrical profile value (can be null)
+     * - power restriction code (can be null)
+     */
+    public record ElecCond(String mode, String profile, String powerRestriction) {
```

Comment:
I'm not a big fan of abbreviation like that, why not `ElectrificationConditions`?

Reply:
I renamed the class, not the variables for concision though (reopen this conversation if you disagree)